test/unit/deferred.js

```diff
@@ -525,6 +525,34 @@ QUnit.test( "jQuery.Deferred.then - spec compatibility", function( assert ) {
 	} catch ( _ ) {}
 } );
 
+QUnit[ window.console ? "test" : "skip" ]( "jQuery.Deferred.exceptionHook", function( assert ) {
+
+	assert.expect( 1 );
+
+	var done = assert.async(),
+		defer = jQuery.Deferred(),
+		oldWarn = window.console.warn;
+
+	window.console.warn = function( msg ) {
```

Comment:
Hm, weird, sinon should be useful in cases like this, especially that it provides integration with assertion libraries like standart `assert` module and `chai`. See - http://chaijs.com/plugins/sinon-chai

@leobalter is there integration way with qunit?